### PR TITLE
fix(agw): Replaced the key for guti hashtable, currently the key is GUTI in string format

### PR DESCRIPTION
**Pull request by rsarwad**
## Summary
Guti hash table is of object hash table, for which earlier key was pointer to object of structure, guti_t. As part of this PR, we changed the key as string format of GUTI. 
Partially fixes the issue, https://github.com/magma/magma/issues/11189
The other part would be fixed by PR, https://github.com/magma/magma/pull/12573
## Test Plan
1. Executed s1ap sanity test suite and manually verified the GUTI values with mme restart, so that GUTI is correctly written to DB and fetched from DB
2. Executed unit test cases.

### CODE REVIEW

File: lte/gateway/c/core/oai/common/conversions.c [hunk 1]
```diff
@@ -367,3 +367,12 @@ char* bytes_to_hex(char* byte_array, int length, char* hex_array) {
   }
   return hex_array;
 }
+
+void convert_guti_to_string(guti_t* guti_p, char (*guti_str)[]) {
+#define GUTI_STRING_LEN 21
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/common/conversions.c]: This is already defined in `conversions.h` below

File: lte/gateway/c/core/oai/tasks/nas/emm/emm_data.h [hunk 2]
```diff
@@ -541,8 +541,6 @@ struct emm_context_s* emm_context_get(emm_data_t* emm_data,
                                       const mme_ue_s1ap_id_t ue_id);
 struct emm_context_s* emm_context_get_by_imsi(emm_data_t* emm_data,
                                               imsi64_t imsi64);
-struct emm_context_s* emm_context_get_by_guti(emm_data_t* emm_data,
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/tasks/nas/emm/emm_data.h:544]: 👍

File: lte/gateway/c/core/oai/tasks/nas/emm/emm_data_ctx.c [hunk 3]
```diff
@@ -662,26 +662,6 @@ struct emm_context_s* emm_context_get_by_imsi(
   return emm_context_p;
 }
 
-//------------------------------------------------------------------------------
-struct emm_context_s* emm_context_get_by_guti(emm_data_t* emm_data,
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/tasks/nas/emm/emm_data_ctx.c:666]: Thanks for the clean up!

**Review by ssanadhya — APPROVED** [review]
Lgtm, with minor change

File: lte/gateway/c/core/oai/common/conversions.c (hunk 1 above)
Comment (rsarwad) [reply lte/gateway/c/core/oai/common/conversions.c]: Thank you.

**Pull request merged**